File: crates/ctxctl/tests/fixtures/sample.lua
local json = require "json"
local helpers = require("./helpers")

-- A counter.
local Counter = {}
Counter.__index = Counter

function Counter.new()
  return setmetatable({ n = 0 }, Counter)
end

-- Adds two numbers.
local function add(a, b)
  return a + b
end

function Counter:increment()
  self.n = self.n + 1
end

local MAX_RETRIES = 3

File: crates/ctxctl/tests/fixtures/sample.c
#include <stdio.h>
#include "util.h"

#define MAX_RETRIES 3

/** A 2D point. */
typedef struct Point {
    double x;
    double y;
} Point;

/** Adds two numbers. */
int add(int a, int b) {
    return a + b;
}

enum Color { RED, GREEN };

static int helper(int v) {
    return v * 2;
}
